cobc/scanner.l

```diff
@@ -1310,12 +1316,51 @@ error_literal (const char *type, const char *literal, unsigned int literal_error
 	cb_error ("%s", err_msg);
 }
 
+/* Scans a symbolic EBCDIC character given as a sequence of decimal digits
+   starting with `c`, and returns the corresponding code as per the machine
+   character set (ASCII or EBCDIC).  */
+static cob_u8_t
+scan_ebcdic_char (int c)
+{
+	char buff[10];	       /* Arbitrary limit, mostly for error-reporting */
+static  cob_u8_t ebcdic_to_ascii[256] ;
+static  int ebcdic_to_ascii_initialized = 0 ;
+
+	unsigned int j = 0;
+	do {
+		buff[j++] = c;
+	} while ((c = input ()) != EOF && '0' <= c && c <= '9' && j < 10);
+	unput (c);
+	buff[j++] = 0;
+	c = atoi (buff);
+	if (c < 1 || c >= 257) {
+		cb_error (_("symbolic character %s out of allowed EBCDIC range (1...256)"),
+			  buff);
+	}
+        c--; // in ordinal, i.e. 1..256 -> 0..255
+#ifdef COB_EBCDIC_MACHINE
+	return (cob_u8_t) c;
+#else
+	if( !ebcdic_to_ascii_initialized ){
+		if (cob_load_collation (cb_ebcdic_table,
+                                        ebcdic_to_ascii,
+                                        NULL) < 0) {
+                        cobc_err_exit (_("invalid parameter: %s"), "-febcdic-table");
```

Comment:
Can we get that error here? If not then please use `(void)cob_load_collation` here, in any case move `ebcdic_to_ascii_initialized = 1;` above because we won't have a different result when getting here the next time and really want to do the loading only once.

Reply:
From the tests that I added following your advises, it looks like we can indeed get the error and exit there. Anyway, I swapped the `ebcdic_to_ascii_initialized = 1;` to always be done.

Reply:
Let's just cb_error + memset the table as '?' to let the program parse until end (because of the swapped setting we won't get in there anymore and just interpret everything as '?''.

Reply:
`cb_error` done, I used a negative value for `ebcdic_to_ascii_initialized` instead of `memset` to return `'?'`